### ISSUE TITLE
The Flutter tool doesn't print out URI for DevTools in web profile mode

### ISSUE THREAD

**Issue by InMatrix** [flutter/flutter#80334]
I was trying to use Dart DevTools to profile the default starter app in Chrome. However, when I ran it in profile mode, the Flutter tool didn't print out a DevTools URI. 
```
➜  new_flutter_app flutter run --profile -d Chrome          
Launching lib/main.dart on Chrome in profile mode...
Compiling lib/main.dart for the Web...                             18.9s

🔥  To hot restart changes while running, press "r" or "R".
For a more detailed help message, press "h". To quit, press "q".
```

In debug mode, the tool didn't print out a DevTools URI either, but at least it gave me the debug service URI. 

```
➜  new_flutter_app flutter run -d Chrome   
Launching lib/main.dart on Chrome in debug mode...
Waiting for connection from debug service on Chrome...             13.2s
Debug service listening on ws://127.0.0.1:58917/MrJ0oUXU3W8=/ws

Running with unsound null safety
For more information see https://dart.dev/null-safety/unsound-null-safety

🔥  To hot restart changes while running, press "r" or "R".
For a more detailed help message, press "h". To quit, press "q".
```

**Comment by jacob314** [comment]
Fyi @jonahwilliams @kenzieschmoll 
 A related feature request would be to have an option to launch in profile mode starting paused with devtools connected at startup so taht devtools can start instrumenting the applications performance and then resume.

**Comment by jonahwilliams** [comment]
There is no vm service to connect to in web profile mode. While we could definitely launch devtools, I'm not sure how useful it would be.

We do need to fix the debug mode launch though

**Comment by jacob314** [comment]
I missed the bug was for web only. For web profile mode we should display a line explain that you need to use chrome devtools to profile at the moment so users don't spend time looking for how to launch Dart DevTools when it wouldn't do any good.

**Comment by InMatrix** [comment]
> For web profile mode we should display a line explain that you need to use chrome devtools to profile at the moment

I didn't know that. So what is the recommended performance debugging workflow for Flutter web apps? It seems that I would need to take at least the following steps:

1. Run my web app in profile mode
2. Measure my app's performance using Chrome DevTools to identify where things are slow.
3. Switch my app to the debug mode
4. Use Dart DevTools to find information that might help me understand why my app was slow
5. Fix the problem in IDE
6. Do 1 & 2 again to verify if the problem is gone

Do the steps above sound about right?

**Comment by jacob314** [comment]
That is correct. The only other option is to profile your app on Flutter desktop to find performance issues hoping that the performance of desktop and web are consistent enough that if you make the desktop performance great, the web performance will be good enough. If you are using the CanvasKit backend, that might be reasonable.

**Comment by InMatrix** [comment]
Do we have plans to support web profile build in Dart DevTools? There are many places the user could get lost in the current workflow.

**Comment by jacob314** [comment]
We've started discussing it but it isn't scheduled yet. To be clear, the issue isn't DevTools but making https://pub.dev/packages/dwds understand Dart2JS output. Current dwds only works with the Dart Dev Compiler used for web debug builds.
